nipy/labs/spatial_models/mroi.py

```diff
@@ -700,14 +700,14 @@ def select_roi(self, id_list):
         # set new features
         # (it's ok to do that after labels and id modification since we are
         # poping out the former features and use the former id indices)
-        for fid in self.features:
+        for fid in list(self.features):
             f = self.remove_feature(fid)
             sf = [f[id] for id in id_list_pos]
```

Comment:
`id_list_pos` is a list of "positions" of the dictionnary elements that should be looped over. This code should break if the order of the dictionnary elements cannot be relied upon.